src/telemetry/logging.ts

```diff
@@ -21,22 +21,15 @@ import { MessageContext, SerializedError } from "@/core";
 import { serializeError } from "serialize-error";
 import { isExtensionContext } from "@/chrome";
 
-function selectError(exc: unknown): SerializedError {
-  if (exc instanceof Error) {
-    return serializeError(exc);
-  } else if (typeof exc === "object") {
-    const obj = exc as Record<string, unknown>;
-    if (obj.type === "unhandledrejection") {
-      return serializeError({
-        // @ts-ignore: OK given the type of reason on unhandledrejection
-        message: obj.reason?.message ?? "Uncaught error in promise",
-      });
-    } else {
-      return serializeError(obj);
-    }
-  } else {
-    return serializeError(exc);
+function selectError(exc: any): SerializedError {
+  if (exc?.message && exc?.type === "unhandledrejection") {
```

Comment:
The previous condition for this was equivalent to:
```suggestion
  if (!(exc instanceof Error) && exc?.type === "unhandledrejection") {
```

But I think that maybe this makes more sense as sets the message only if it doesn't already exist